components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -757,6 +757,14 @@ void AdsServiceImpl::MaybeStart(const bool should_restart) {
     return;
   }
 
+  if (is_start_in_progress_) {
+    VLOG(1) << "Previous ads service start is in progress.";
+    return;
+  }
+  // Do not allow to do more Start() calls until
+  // OnEnsureBaseDirectoryExists() is called.
+  is_start_in_progress_ = true;
```

Comment:
Should we not set `is_start_in_progress_` to `false` if the service is stopped in `Stop()`?

Reply:
There is no need to set is_start_in_progress_ to false because OnEnsureBaseDirectoryExists() is always executed if Start() was called.
I have refactored this fix to be more readable.